src/js/structs/Application.js

```diff
@@ -7,57 +9,28 @@ import TaskStats from './TaskStats';
 import VolumeList from './VolumeList';
 
 module.exports = class Application extends Service {
-  getAcceptedResourceRoles() {
-    return this.get('acceptedResourceRoles');
-  }
 
-  getArguments() {
-    return this.get('args');
-  }
-
-  getCommand() {
-    return this.get('cmd');
-  }
-
-  getContainerSettings() {
-    return this.get('container');
-  }
-
-  getCpus() {
-    return this.get('cpus');
-  }
-
-  getContainer() {
-    return this.get('container');
-  }
-
-  getConstraints() {
-    return this.get('constraints');
-  }
+  // getContainer() {
+  //   return this.get('container');
+  // }
 
   getDeployments() {
     return this.get('deployments');
   }
 
-  getDisk() {
-    return this.get('disk');
-  }
-
-  getEnvironmentVariables() {
-    return this.get('env');
-  }
-
-  getExecutor() {
-    return this.get('executor');
-  }
-
+  /**
+   * @override
+   */
   getSpec() {
-    // DCOS-9613: This should be properly implemented
-    return this;
+    return new ApplicationSpec(cleanServiceJSON(this.get()));
```

Comment:
Let's move the clean json part to the spec, so that we can creates specs directly from the server response.